fedimint-client/src/transaction/sm.rs

```diff
@@ -154,44 +124,47 @@ impl IntoDynInstance for TxSubmissionStates {
     }
 }
 
-async fn trigger_created_submit(
+async fn trigger_created_rejected(
     tx: Transaction,
-    next_submission: SystemTime,
     context: DynGlobalClientContext,
-) -> Result<TransactionId, String> {
-    fedimint_core::task::sleep(
-        next_submission
-            .duration_since(now())
-            .unwrap_or(Duration::ZERO),
-    )
-    .await;
-
-    // TODO: get rid of state machine created->created loop and only rely on this
-    // loop
+) -> Result<String, String> {
     loop {
         match context.api().submit_transaction(tx.clone()).await {
-            Err(e) if e.is_retryable() => {
-                debug!("Got {e} while submitting transaction, will sleep for {RESUBMISSION_INTERVAL:?}");
-                sleep(RESUBMISSION_INTERVAL).await;
+            Ok(submission_result) => {
+                if let Err(submission_error) = submission_result {
+                    return Ok(submission_error.to_string());
+                }
+            }
+            Err(error) => {
+                if !error.is_retryable() {
```

Comment:
Any reason to not use `Err(e) if e.is_retryable()` anymore?

Reply:
I find match guards in this case less intuitive and early return here more idiomatic.